### PR TITLE
Add simplistic editor to samples to showcase TextEditingDeltas

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
This PR adds a simple text editor sample that showcases `TextEditingDeltas` and the `DeltaTextInputClient` by using the information from deltas to expand and contract styled ranges of text.

https://user-images.githubusercontent.com/948037/166822126-fd2690f0-ec6c-49e4-b12e-2319faa67a2a.mov

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [] All existing and new tests are passing.

### CODE REVIEW

**Comment by domesticmouse** [comment]
Please add this project to CI and Pub Dependabot:
  - [tool/flutter_ci_script_stable.sh](https://github.com/flutter/samples/blob/master/tool/flutter_ci_script_stable.sh)
  - [tool/flutter_ci_script_beta.sh](https://github.com/flutter/samples/blob/master/tool/flutter_ci_script_beta.sh)
  - [.github/dependabot.yaml](https://github.com/flutter/samples/blob/master/.github/dependabot.yaml)

**Review by craiglabenz — APPROVED** [review]
This looks like an awesome demo!

My current thoughts are two-fold _(update from the future: pseudo-three-fold, but not really):_

1 - I would love to see a meatier README that gives me some guidance on how to consume this code and how to mentally separate what is only needed for your larger visualization layer versus what I should expect to need for any use of the new text delta APIs, and

2 - On a much more narrow note - I noticed that SHIFT-<arrow keys> and OPTION-<arrow keys> did not seem to have any effect. Is that a bug? A thing that works but is outside the scope of this demo? Something else? Ideally, I think it would be nice if those forms of text selection functional in this demo.

3 - Overall, I love it! I'm going to give this a LGTM so we can get it landed and produce a permalink, but I'm definitely invested in the README enhancements :)

**Comment by craiglabenz** [comment]
@Renzo-Olivares I gave this a LGTM, but that was without realizing that the minimum Dart SDK version will _have_ to be set to `2.17.0-0` before we can successfully integrate this into https://flutter.github.io/samples.

**Comment by Renzo-Olivares** [comment]
> @Renzo-Olivares I gave this a LGTM, but that was without realizing that the minimum Dart SDK version will _have_ to be set to `2.17.0-0` before we can successfully integrate this into [flutter.github.io/samples](https://flutter.github.io/samples).

Thanks for the review @craiglabenz I will definitely get to working on a beefier README to better describe the structure of this sample, and separate the visualization layer from the core delta layer. Also I have updated the minimum Dart SDK in the pubspec.

In regards to the arrows keys, there are many shortcuts that weren't defined because `DefaultTextEditingShortcuts` and their corresponding `Actions` are not reusable at the moment (This is being worked on this quarter). I wanted to avoid just copy and pasting private implementations into the demo to keep it as simple as possible. I agree though that it would be great to have the shift + arrow working at least, so I'll look into what I can do about those.

File: simplistic_editor/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# build by specifying --build-name and --build-number, respectively.
+# In Android, build-name is used as versionName while build-number used as versionCode.
+# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
+# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
+# Read more about iOS versioning at
+# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
+version: 1.0.0+1
+
+environment:
+  sdk: '>=2.18.0-63.0.dev <3.0.0'
```
[the remark this answers is not in the document]
Comment (Renzo-Olivares) [reply simplistic_editor/pubspec.yaml]: Done

File: simplistic_editor/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# simplistic_editor
+
+This sample text editor showcases the use of TextEditingDelta's and a DeltaTextInputClient to expand
+and contract styled ranges of text. For more information visit https://api.flutter.dev/flutter/services/TextEditingDelta-class.html.
```
[the remark this answers is not in the document]
Comment (Renzo-Olivares) [reply simplistic_editor/README.md:4]: This makes sense to me! I'll try to demystify this sample as much as possible in the README.

File: simplistic_editor/README.md [hunk 3]
```diff
@@ -4,81 +4,85 @@ and contract styled ranges of text. For more information visit https://api.flutt
 
 # Structure
 ## Visualization Layer
-The layer that showcases the TextEditingDelta history in a ListView.
+The layer that showcases the `TextEditingDelta` history in a ListView. These widgets are all unique to this sample
+and are unlikely to contain much logic the typical user will need.
```
Comment (craiglabenz) [inline simplistic_editor/README.md:10]: Added this sentence:

  > These widgets are all unique to this sample and are unlikely to contain much logic the typical user will need.

Can you confirm that this is accurate, @Renzo-Olivares?

File: simplistic_editor/README.md [hunk 4]
```diff
@@ -4,81 +4,85 @@ and contract styled ranges of text. For more information visit https://api.flutt
...
+### `TextEditingDeltaView`
+A widget that represents the content of a `TextEditingDelta`. A list of `TextEditingDeltaView`s sits below
+the input field showcasing a history of `TextEditingDelta`s that have occurred on that input field. A
+`TextEditingDeltaView` varies in color depending on the type of `TextEditingDelta`. `TextEditingDeltaInsertion`s
+are green, `TextEditingDeltaDeletion`s are red, `TextEditingDeltaReplacement`s are yellow, and `TextEditingDeltaNonTextUpdate`s are blue.
 
 ## Replacements Layer
 The layer that handles the styling of the input field, including expanding and contracting the styled ranges based
 on the deltas that are received from the platform, and the handling of the state of the styling toggle button toolbar.
+This layer contains a mixture of logic unique to this sample and helpful for developers expecting to consume the
```
Comment (craiglabenz) [inline simplistic_editor/README.md:33]: Added this sentence:

> This layer contains a mixture of logic unique to this sample and helpful for developers expecting to consume the
`TextEditingDelta` APIs.

Can you confirm that this is correct, @Renzo-Olivares?

File: simplistic_editor/README.md [hunk 5]
```diff
@@ -4,81 +4,85 @@ and contract styled ranges of text. For more information visit https://api.flutt
...
 
 This controller also contains convenience methods used by the styling toggle buttons toolbar to un-style
 certain ranges, disable the expand property of a replacement, and get the common replacements at a selection
 to determine the current toggle button state.
 
 ## Text Input Layer
-The layer that defines the appearance of a text input field, handles the text input received from 
-the platform, and mutations done by the framework through gestures and keyboard shortcuts.
+The layer that defines the appearance of a text input field, handles the text input received from
+the platform, and mutations done by the framework through gestures and keyboard shortcuts. These classes begin to
```
Comment (craiglabenz) [inline simplistic_editor/README.md:70]: Added this sentence:

> These classes begin to demonstrate the types of logic developers may need if they wish to interact with `TextEditingDelta`s.

Can you confirm that this is correct, @Renzo-Olivares?

File: simplistic_editor/pubspec.lock [hunk 6]
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.9.0"
+    version: "2.8.2"
```
Comment (craiglabenz) [inline simplistic_editor/pubspec.lock:10]: weird 🤔

File: simplistic_editor/README.md (hunk 3 above)
Comment (Renzo-Olivares) [reply simplistic_editor/README.md:10]: Accurate!

File: simplistic_editor/README.md (hunk 5 above)
Comment (Renzo-Olivares) [reply simplistic_editor/README.md:70]: This is correct!

**Comment by Renzo-Olivares** [comment]
@craiglabenz I'm unsure why some of these tests are failing. Do you have any suggestions?

**Comment by craiglabenz** [comment]
It looks like this requires a quick linting pass - I can help with that.

**Pull request merged**